tensorboard/plugins/scalar/vz_line_chart/vz-line-chart.ts

```diff
@@ -68,14 +72,35 @@ Polymer({
     smoothingWeight: {type: Number, value: 0.6},
 
     /**
-     * The way to display the X values. Allows:
-     * - "step" - Linear scale using the  "step" property of the datum.
-     * - "wall_time" - Temporal scale using the "wall_time" property of the
-     * datum.
-     * - "relative" - Temporal scale using the "relative" property of the
-     * datum if it is present or calculating from "wall_time" if it isn't.
+     * We accept a function for creating an XComponents object instead of such
+     * an object itself because the Axis must be made right when we make the
+     * LineChart object, lest we use a previously destroyed Axis. See the async
+     * logic below that uses this property.
+     * @type {function(): ChartHelpers.XComponents}
      */
-    xType: {type: String, value: 'step'},
+    xComponentsCreationMethod: Object,
+
+    /**
+     * A method that implements the Plottable.IAccessor<number> interface. Used
+     * for accessing the y value from a data point.
+     */
+    yValueAccessor: Object,
+
+    /**
+     * An array of ChartHelper.TooltipColumn objects. Used to populate the table
+     * within the tooltip. The table contains 1 row per run.
+     */
+    tooltipColumns: Array,
+
+    /**
+     * Tooltip header innerHTML text. We cannot use a dom-repeat inside of a
+     * table element ... because polymer is awry. Hence, we manually generate
```

Comment:
Could you replace "because Polymer is awry" with something like, "because Polymer does not support this" or "due to a Polymer bug" or something? (_Awry_ is pretty non-specific here.)